railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -52,6 +52,10 @@ def start
         ENV["PGSSLCERT"]      = config[:sslcert].to_s if config[:sslcert]
         ENV["PGSSLKEY"]       = config[:sslkey].to_s if config[:sslkey]
         ENV["PGSSLROOTCERT"]  = config[:sslrootcert].to_s if config[:sslrootcert]
+        if config[:variables]
+          variables = config[:variables].reject { |_k, v| v == ":default" || v == :default }
+          ENV["PGOPTIONS"] = variables.map { |k, v| "-c #{k}=#{v.gsub(" ", "\\ ").gsub("\\", "\\\\")}" }.join(" ")
```

Comment:
I don't think the `gsub` calls will work as expected:

```ruby
"escape \\this".gsub(" ", "\\ ").gsub("\\", "\\\\")
# => "escape\\ \\this"
```

But the following should work:

```ruby
"escape \\this".gsub(/[ \\]/, '\\\\\0') # MOAR BACKSLASHES!!!
# => "escape\\ \\\\this"
```

Perhaps we should add a value with a backslash in the test?

Also, what do you think about the following refactor?

```suggestion
          ENV["PGOPTIONS"] = config[:variables].filter_map do |name, value|
            "-c #{name}=#{value.gsub(/[ \\]/, '\\\\\0')}" unless value == ":default" || value == :default
          end.join(" ")
```

Reply:
Thanks! I forgot about `filter_map` and often have troubles with escaping 😄